File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/MoneyConsumer.java
package combined.avro_events.consumer;

import com.example.events.common.Money;
import combined.avro_events.header.StandardHeaders;
import java.lang.AutoCloseable;
import java.lang.Void;
import java.time.Duration;
import java.util.concurrent.CompletableFuture;
import java.util.stream.StreamSupport;
import org.apache.kafka.clients.consumer.Consumer;
import org.apache.kafka.clients.consumer.ConsumerRecords;

/** Type-safe consumer for money topic */
public record MoneyConsumer(
  Consumer<String, Money> consumer,
  MoneyHandler handler,
  String topic
) implements AutoCloseable {
  public MoneyConsumer(
    Consumer<String, Money> consumer,
    MoneyHandler handler
  ) {
    this(consumer, handler, "money");
  }

  public MoneyConsumer withConsumer(Consumer<String, Money> consumer) {
    return new MoneyConsumer(consumer, handler, topic);
  }

  public MoneyConsumer withHandler(MoneyHandler handler) {
    return new MoneyConsumer(consumer, handler, topic);
  }

  public MoneyConsumer withTopic(String topic) {
    return new MoneyConsumer(consumer, handler, topic);
  }

  /** Close the consumer */
  @Override
  public void close() {
    consumer.close();
  }

  /** Poll for messages and dispatch to handler, returning composed effect */
  public CompletableFuture<Void> poll(Duration timeout) {
    ConsumerRecords<String, Money> records = consumer.poll(timeout);
    return CompletableFuture.allOf(StreamSupport.stream(records.spliterator(), false).map(record -> handler.handle(record.key(), record.value(), StandardHeaders.fromHeaders(record.headers()))).toArray(CompletableFuture[]::new));
  }
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/MoneyConsumer.kt
package combined.avro_events.consumer

import com.example.events.common.Money
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Multi
import io.smallrye.mutiny.Uni
import java.io.Closeable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for money topic */
data class MoneyConsumer(
  val consumer: Consumer<kotlin.String, Money>,
  val handler: MoneyHandler,
  val topic: kotlin.String = "money"
) : Closeable {
  /** Close the consumer */
  override fun close() {
    consumer.close()
  }

  /** Poll for messages and dispatch to handler, returning composed effect */
  fun poll(timeout: Duration): Uni<Unit> {
    val records: ConsumerRecords<kotlin.String, Money> = consumer.poll(timeout)
    return Multi.createFrom().iterable(records).onItem().transformToUniAndConcatenate({ record -> handler.handle(record.key(), record.value(), StandardHeaders.fromHeaders(record.headers())) }).collect().last()
  }
}